### PR TITLE
PREAPPS-1038: Add `tokenType` option to `AuthRequest`

### PR DESCRIPTION
**Pull request by pl12133**
Also pass on response of `AuthRequest` instead of resolving with boolean.

### CODE REVIEW

File: src/schema/generated-schema-types.ts [hunk 1]
```diff
@@ -977,6 +977,26 @@ export interface NameId {
 	name?: string | null;
 }
 
+export interface AuthResponse {
+	authToken?: AuthToken[] | null;
```
Comment (silentsakky) [inline src/schema/generated-schema-types.ts]: usually auth token is stored as cookie only, we should not store it somewhere else

Comment (pl12133) [reply src/schema/generated-schema-types.ts]: I think you are right and this is actually already a problem, see https://github.com/apollographql/apollo-client/issues/3592.

It looks like we should likely be using `fetchPolicy: 'no-cache'` for the `AuthRequest` endpoint to workaround the issue.

Comment (plobbes) [reply src/schema/generated-schema-types.ts]: Thanks @silentsakky and @pl12133.  I agree we should not be storing/caching the auth tokens here.

Comment (silentsakky) [reply src/schema/generated-schema-types.ts]: or maybe if authtoken is not used anywhere in the application then remove it from schema? that way we will not be storing in cache at all

**Pull request merged**